src/Microsoft.ApplicationInsights.AspNetCore/Extensions/ApplicationInsightsExtensions.cs

```diff
@@ -44,8 +46,14 @@ public static void AddApplicationInsightsTelemetry(this IServiceCollection servi
         {
             var options = serviceOptions ?? new ApplicationInsightsServiceOptions();
 
+            services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
+            services.AddSingleton<IActionContextAccessor, ActionContextAccessor>();
+
             services.AddSingleton<ITelemetryInitializer, DomainNameRoleInstanceTelemetryInitializer>();
-            services.AddSingleton<ITelemetryInitializer, ComponentVersionTelemetryInitializer>();
+            services.AddSingleton<ITelemetryInitializer>(serviceProvider =>
+            {
+                return new ComponentVersionTelemetryInitializer(config);
```

Comment:
ImplementationType will be null again. But, seems like this is a more straightforward way to proceed.